File: kaiburr-task3-ui/src/api/TaskService.ts
import axios from 'axios';
import { Task } from '../types/task';


const BASE_URL = 'http://localhost:8080';
 // Your Java backend URL

// Create a new task
export const createTask = (task: Task) => {
  return axios.post(`${BASE_URL}/tasks`, task);
};

// Get tasks by name (for search functionality)
export const getTaskByName = (name: string) => {
  return axios.get<Task[]>(`${BASE_URL}/tasks?name=${name}`);
};

File: kaiburr-backend-task3/src/main/java/com/kaiburr/taskmanager/service/TaskService.java
package com.kaiburr.taskmanager.service;

import com.kaiburr.taskmanager.model.Task;
import com.kaiburr.taskmanager.repository.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class TaskService {

    @Autowired
    private TaskRepository taskRepository;

    // Create or update a task
    public Task createTask(Task task) {
        return taskRepository.save(task);
    }

    // Search for tasks by name
    public List<Task> getTasksByName(String name) {
        return taskRepository.findByNameContainingIgnoreCase(name);
    }

    // Get all tasks
    public List<Task> getAllTasks() {
        return taskRepository.findAll();
    }
}
